include/ignition/gui/GuiEvents.hh

```diff
@@ -449,6 +450,34 @@ namespace ignition
         /// \brief Private data pointer
         IGN_UTILS_IMPL_PTR(dataPtr)
       };
+
+      /// \brief Event which is called to share WorldControl information.
+      class IGNITION_GUI_VISIBLE WorldControl : public QEvent
+      {
+        /// \brief Constructor
+        /// \param[in] _worldControl The WorldControl information
+        /// \param[in] _play Whether the event is associated with a user
+        /// pressing play (true) or not (false)
+        public: WorldControl(const msgs::WorldControl &_worldControl,
+                    bool _play);
+
+        /// \brief Unique type for this event.
+        static const QEvent::Type kType = QEvent::Type(QEvent::MaxUser - 19);
+
+        /// \brief Get the WorldControl information
+        /// \return The WorldControl information
+        public: const msgs::WorldControl &WorldControlInfo() const;
+
+        /// \brief Get whether the event is associated with
+        /// a user selecting "play" or not
+        /// \return True if the event is associated with a user selection play.
+        /// False otherwise
```

Comment:
I think this could be a bit clearer. The first time I read it, I thought it would return the same as `WorldControlInfo().pause()` (i.e. false may mean pause or no action). But looking at the code, false here does mean pause. So I think it's worth documenting it:

```suggestion
        /// False if the user selected "pause".
```

Reply:
False can mean more than just pause - for example, in this PR, false is also used for step: https://github.com/ignitionrobotics/ign-gui/blob/915a1a6fda3cbbd2fdb0a4af3238520e7bc53a7c/src/plugins/world_control/WorldControl.cc#L288

So, what if I said something like: `False if the user did not select play (some examples might be the user selecting "pause" or "step")`.

I don't know if we should document that false can only mean "pause", because perhaps downstream users have other interpretations of what false could be.

Reply:
> for example, in this PR, false is also used for step: 

Humm but the meaning is still that simulation should not continue running after receiving this command, right? I don't think that's different from saying that means pause :thinking:

Reply:
So, after addressing https://github.com/ignitionrobotics/ign-gazebo/pull/1109#discussion_r740421855, this event actually does not need the `Play` method anymore. `ign-gazebo` has everything it needs from just the `msgs::WorldControl` piece. So, I've removed the `Play` method in https://github.com/ignitionrobotics/ign-gui/pull/306/commits/7ff16ce47ffbb1e0dc37d005333a4111a13e505d. I'll mark this as resolved, but feel free to re-open it if you have any more comments/questions.